[ui-2.6] New Volume Page (To be merged)

ChengYanJin opened the pull request:
**Component** ui, volumes, monitoring

**Context**:  Replace the old volume information page with the new Volume Page

**Summary**:
- Volume list is implemented by `react-table`, 
① the selected row is highlighted so that we can clearly know which volume we are looking for)
② store the search bar content in `localStorage`, so we won't lose them during the refresh and API call.
③ add scroll bar 

- Volume Detail Card ( Note that I remove the Volume Health Status Bar since we can't get the history of the alerts yet)
- Volume Active Alert Card ( Note that I still get the alerts through `Prometheus` not `AlertManager` yet, the alert of AertManager has a weird behavior we need a further investigation on it, please see #2662)
- Metrics Chart Card 
① We can change the time span from the dropdown, by default is `last 7 days`
② rename it from PERFORMANCE to METRICS
③  add auto-refreshing( refresh the metrics graph every 1 min)

- Add a node field in Create Volume Page.
- Update the URL and Breadcrumb

URL for volume page:
`/volumes`: Access the volume page same as from the sidebar
`/volumes/<volume-name>`: Check the detail of volume <volume-name> 
`/volumes/?node=<node-name>`: Add the node filter, if we go to the volume page from node page
`/volumes/createVolume`: Create a volume from the `/volumes` page
`/volumes/createVolume/?node=<node-name>`: Create a volume from with node filter, the `node` field prefilled by node filter

We don't want to re-render the whole VolumePage when we select to another volume
How to design the components: 

- `VolumePage` component (smart)
     - get the list of volumes
- `VolumeContent` component (smart)
    - extract the volume from the URL
    - tell the sub-router component the selected volume
    - get the metric data from monitoring ( for the MetricChart)
- `VolumeListTable` component (dumb)
    - get the volumelist and selected volume through props 
- `RightSidePanel`component
   - `VolumeDetailCard` component (dumb)    
   - `ActiveAlertCard`  component (dumb)
   - `MetricGraphCard` component (dumb)

We design those three components as a dumb component is because we may want to add a tab to switch between those components. And we don't want them to be refreshed. 

**Acceptance criteria**: 
Dark theme:
![image](https://user-images.githubusercontent.com/18453133/90504216-0b83a980-e151-11ea-8e79-87ed47f17762.png)
![image](https://user-images.githubusercontent.com/18453133/90504379-42f25600-e151-11ea-92ad-d025892b383c.png)
![image](https://user-images.githubusercontent.com/18453133/90232192-921c4c00-de1c-11ea-8e64-ce69f4cbd291.png)
Light theme:
![image](https://user-images.githubusercontent.com/18453133/90509524-bd26d880-e159-11ea-8f2a-4c27efbbccbc.png)
![image](https://user-images.githubusercontent.com/18453133/90509547-c6b04080-e159-11ea-95ec-86cd6c8619f2.png)



Note that: 
In order to merge the New Volume Page, I temporarily remove the `CreateVolume.feature` for E2E test. Will add it back in the next coming PR.

How to fill up the volume? take alertmanger volume as example which the size is `10GiB` the devicepath is `/dev/vdb`
1. ssh centos@10.200.5.215
2. cd dev/
3. mount /dev/vdb my-volume
4. dd if=/dev/urandom of=my-volume/bigfile bs=1M count=9500 seek=0


<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2505 
Closes: #2506
Closes: #2507 
Closes: #2508

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui/theme: apply the new color palette

ChengYanJin committed:
ui/components: modify the components to adapt to the new color palette

ChengYanJin committed:
ui/theme: Switch the theme & dex login to dark mode

Refs: #2528

ChengYanJin committed:
ui/color-palette: Merge the color palette

- textSecondary (#A8B5C1) & textTertiary (#A7B6C3) => keep text Secondary #A8B5C1
- info button color (#027AFF) & Secondary color is blue (#037AFF) => > Keep #037AFF which should be the color of the most important action button, and the selection feedback color.

Refs: #2528

ChengYanJin committed:
ui/volumes: Volume Active Alert Card

ChengYanJin committed:
ui/volumes: Volume Detail Card

ChengYanJin committed:
ui/volumes: Volume List Table

ChengYanJin committed:
ui/volumes: Add Volume Metric Graoh Card

ChengYanJin committed:
ui/volumes: Add New Volume Page

ChengYanJin committed:
ui/volumes: Update the translation

ChengYanJin committed:
ui/volumes: Update the core-ui version for volume page

ChengYanJin committed:
ui/volumes: Add the utility functions for Volume Page

ChengYanJin committed:
ui/volumes: Update the breadcrumb styles & layout container

ChengYanJin committed:
ui/volumes: Add Node name field in Create Volume Form

ChengYanJin committed:
ui/volumes: Add the constant volumes for New Volume Page

ChengYanJin committed:
ui/volumes: update the ducks and services for New Volume Page

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

ChengYanJin committed:
ui/volumes: Adapt the new volume page to dev/2.6

ChengYanJin committed:
ui/volumes: Update the package-lock.json

ChengYanJin committed:
ui/volumes: Temporarily Remove the CreateVolume Feature E2E test

The New Volume Page breaks the E2E test, will address it in the coming PR

ChengYanJin committed:
ui/tests: Update the volume unit tests

- We added a node field in create volume which break the unit test

ChengYanJin committed:
ui/volumes: Adapt the light theme to New Volume Page

The Progressbar background color didn't fit with light theme

ChengYanJin committed:
ui/performance: Increase the maxAssetSize which cause the npm error ELIFECYCLE

The error we had is:
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 SIGINT listeners added. Use emitter.setMaxListeners() to increase limit\nBrowserslist: caniuse-lite is outdated. Please run next command `npm update`\nnpm ERR! code ELIFECYCLE

ChengYanJin committed:
ui/volumes: Change the timespan to compute the average latency for volume table

- The query of latency `rate(node_disk_io_time_seconds_total{job="node-exporter"}[1h])`
- Todo: We should give a hint to user that we compute the latency based on the last 1 hour

thomasdanan commented on salt/metalk8s/addons/ui/deployed/ui.sls:
```diff
@@ -59,4 +59,4 @@ Create ui-branding ConfigMap:
         data:
           theme.json: |
-           {"theme": {"light":{"brand":{"base": "#607080", "primary": "#FAF9FB", "primaryDark1": "#F7F6F9", "primaryDark2": "#EDEAF0", "secondary": "#037AFF", "secondaryDark1": "#1C3D59", "secondaryDark2": "#1C2E3F", "success": "#006F62", "healthy": "#25AC56", "healthyLight": "#75FE63", "info": "#027AFF", "warning": "#FFC10A", "danger": "#EF3340", "critical": "#BE2543", "background": "#ffffff", " ...
+           {"theme": {"light":{"brand":{"base": "#607080", "primary": "#FAF9FB", "primaryDark1": "#F7F6F9", "primaryDark2": "#EDEAF0", "secondary": "#037AFF", "secondaryDark1": "#1C3D59", "secondaryDark2": "#1C2E3F", "success": "#006F62", "healthy": "#25AC56", "healthyLight": "#75FE63", "warning": "#FFC10A", "danger": "#EF3340", "critical": "#BE2543", "background": "#ffffff", "backgroundBluer": " ...
```
```suggestion
          {"theme":{"light":{"brand":{"base":"#607080","primary":"#FAF9FB","primaryDark1":"#F7F6F9","primaryDark2":"#EDEAF0","secondary":"#037AFF","secondaryDark1":"#1C3D59","secondaryDark2":"#1C2E3F","success":"#006F62","healthy":"#25AC56","healthyLight":"#75FE63","warning":"#FFC10A","danger":"#EF3340","critical":"#BE2543","background":"#ffffff","backgroundBluer":"#ECF4FF","textPrimary":"#313B44","textSecondary":"#8593A0","borderLight":"#A5A5A5","border":"#A5A5A5"},"logo_path":"/brand/assets/branding-light.svg"},"dark":{"brand":{"base":"#6A7B92","primary":"#1D1D1F","primaryDark1":"#171718","primaryDark2":"#0A0A0A","secondary":"#037AFF","secondaryDark1":"#1C3D59","secondaryDark2":"#1C2E3F","success":"#006F62","healthy":"#25AC56","healthyLight":"#75FE63","warning":"#FFC10A","danger":"#EF3340","critical":"#BE2543","background":"#121214","backgroundBluer":"#182A41","textPrimary":"#FFFFFF","textSecondary":"#A8B5C1","borderLight":"#2C3137","border":"#A5A5A5"},"logo_path":"/brand/assets/branding-dark.svg"},"custom":{}},"default":"dark"}
```

thomasdanan replied:
just removed one wrong curly brace

ChengYanJin committed:
ui/volumes: Fix the bug in configMap ui-branding

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue 2505-2506.

Goodbye chengyanjin.
